Refactor/general

0x4007 opened the pull request:
Resolves #574

Refactoring to clean up the codebase.

Based on #643
- Bounties -> Tasks
- Removed all Telegram bot code

Secrets
`PROJECT_ID` -> `SUPABASE_PROJECT_ID`

I'm not great at the database side. I realized I changed some property names and might have broken the permits database at 03466d6348592006c50731120f4222d457dd1c94

Included most of @whilefoo's end to end tests but a few dont work like `allow` and `start/stop` those are commented out

wannacfuture commented:
should I resolve the conflicts? @pavlovcik

pavlovcik commented:
This should have merged this in weeks ago. The number of outstanding pull requests is growing every month and this is unacceptable. Most of these code changes in this pull request were simple renaming of variables. It would have taken 15 minutes to review and approve @0xcodercrane 

> should I resolve the conflicts? @pavlovcik

I did a pass at it because I know the code I wrote, and I've also been looking at all the pull requests so I have sufficient context. I did it quickly in the GitHub UI though. Please ensure that it builds etc.

pavlovcik commented on yarn.lock:
Just rebuild this because its not worth solving these merge conflicts.

0xcodercrane replied:
but shouldn't delete it from the repo

cuddle-dev reviewed (dismissed):
We shouldn't delete yarn.lock because it's being used in github ci / netlify ci. Due to permission, we couldn't generate it on both platform. That's why we've been keeping it from the beginning.

pavlovcik commented:
@0xcodercrane In this case you should consider just regenerating the lock file and merging instead of kicking the can down the road. I'm not at home so I don't have the capability to from my phone.

0xcodercrane commented:
should I merge https://github.com/ubiquity/ubiquibot/pull/643 in prior to this PR? @pavlovcik

pavlovcik commented:
Don't think it matters the order regarding conflicts but technically speaking this is branched off of #643 head.

0x4007 commented:
Will need to e2e test and merge

0x4007 commented:
Review comments at https://github.com/ubiquity/ubiquibot/pull/849

whilefoo commented on src/bindings/bot-runtime.ts:
```diff
@@ -0,0 +1,60 @@
+import { createAdapters } from "../adapters";
+import { Logs } from "../adapters/supabase";
+
+class Runtime {
```
this will still cause the issue described in #732 because there's only one instance of Runtime and multiple requests will override each other. this is fine for adapters and logger because they should only be initialized once, but event context and config should not be global/singleton

0x4007 replied:
I'm working on your issue [now](https://github.com/pavlovcik/ubiquibot/compare/refactor/general...pavlovcik:ubiquibot:feat/event-context-array?expand=1), because I noticed it as well. Although I'm tempted to work on it in another pull request.

rndquu commented:
@pavlovcik 

There is the supabase [diff tool](https://supabase.com/docs/reference/cli/supabase-db-diff) which can create "diff" migrations between DB schemas.

So I took your local DB dump (which you DMed me) and created this [migration script](https://github.com/rndquu/bounty-bot/blob/f09ba3a1363c01417fc236b1242a7c32f96603b2/supabase/migrations/20231019155701_refactor_general.sql).

Here is how I generated it:
1. Start supabase locally (migrations are applied automatically on start)
```
yarn supabase start
```
2. Create a remote supabase project at https://supabase.com/dashboard/projects
3. Connect to a remote DB (at supabase) with [pgAdmin](https://www.pgadmin.org/)
4. Restore DB from a sql dump of the updated DB schema. Select the `public` schema in `pgAdmin`, open `psql` tool from `pgAdmin` and run sql dump script via `\i  /Users/username/Downloads/updated_schema.sql`.
5. Link a remote project
```
yarn supabase link -p PASSWORD --project-ref PROJECT_REF
``` 
6. Run the diff tool between a local DB (without changes) and a remote DB from supabase (with updated schema)
```
yarn supabase db diff --linked -f refactor_general
```
7. Apply newly generate "diff" migration
```
yarn supabase db reset
```

This "diff" migration should be generated and debugged when you're finished with the current refactor. So when you're finished you could share a sql dump of your final DB schema and I would generate and debug the "diff" migration.

Tomorrow I will check [supabase triggers](https://supabase.com/docs/guides/database/postgres/triggers) and how we can use them to populate the `locations` (i.e. metadata) table.

rndquu commented:
@pavlovcik 

Regarding the `locations` and metadata.

This is how the schema looks like right now:
<img width="1117" alt="2be1a71c-1239-4033-856a-7586359e9691" src="https://github.com/ubiquity/ubiquibot/assets/119500907/dcd7359d-4808-48b3-90f4-3966bb4b90e7">


Each table has the `location_id` field which is a reference to the `location` entity which is basically some fields from the github event that caused the row to be inserted in the original table. 

We have 3 DB operations which cause an update:
1. create
2. update
3. delete

The issue with the current schema is that it reflects only `create` DB operations. For example, when user registers a new wallet then a new corresponding `location` entity is created and `wallet.location_id` is populated. But when a user updates a wallet then it is not clear what we should do with the `wallet.location_id` field, rewrite it or keep the old value.

It is weird, but supabase does not support insertions with relations which leaves us with 3 options:
1. Use postgres [RPC calls](https://supabase.com/docs/reference/javascript/rpc). So for each model update we should write a separate RPC method which is really hard to a maintain.
2. Use postgres [triggers](https://supabase.com/docs/guides/database/postgres/triggers). This approach adds redundancy because for each table we should add a new virtual field `location` which we should keep in mind on all create/update/delete operations (ex: `await supabase.from("wallets").upsert({...walletObj, ...locationObj}`)
3. Use 2 separate API calls

The original issue sounds like "we should save github event metadata to all DB create/update/delete operations". We could do it this way:
1. Remove `location_id` from all of the tables
2. Add `uuid` field to all of the tables
3. Modify the `locations` table to have the following fields: `id serial`, `entity_uuid uuid`, `operation text`, `github_event jsonb`
4. Hook into the original `supabase client SDK` to send a 2nd API call to the `locations` table on each create/update/delete operation

## Example (user registers a new wallet):
### wallet
| uuid  | address |
| ------------- | ------------- |
| 1111-2222  | 0x01  |
### locations
| id  | entity_uuid | operation | github_event |
| ------------- | ------------- | --- | --- |
| 1  | 1111-2222  | create | `{node_id:1,node_html:"https://github.com",issue_id:3,etc...}` |

## Example (user updates a wallet):
### wallet
| uuid  | address |
| ------------- | ------------- |
| 1111-2222  | 0x02  |
### locations
| id  | entity_uuid | operation | github_event |
| ------------- | ------------- | --- | --- |
| 1  | 1111-2222  | create | `{node_id:1,node_html:"https://github.com",issue_id:3,etc...}` |
| 2  | 1111-2222  | update | `{node_id:2,node_html:"https://github.com",issue_id:3,etc...}` |

Overall this issue of tracking corresponding metadata (i.e. github event context object) can be safely moved to a separate issue in order not to blow the current PR.

0x4007 commented:
There are many dimensions of your proposal to consider so it's hard for me to choose with confidence. However based on the existing schema, I think that an RPC call is the right move. 

Alternatively we do the two API calls but there's a good chance that one or the other might fail which can lead to an inconsistent database state. As long as we have a maintainable solution for that, then we can do this approach as it might be simpler.

rndquu commented:
> Alternatively we do the two API calls but there's a good chance that one or the other might fail

We can start simple for now and then implement some retry strategy for failed metadata updates.

0x4007 commented:
This should work on personal accounts (not just organizations) as well now. I forgot to mention I implemented that a week or two ago.

0x4007 committed:
refactor: cleanup

0x4007 committed:
fix: filepath

0x4007 committed:
chore: init refactor bot config

0x4007 committed:
refactor(config): ajv broken cant figure out what the error message means

0x4007 committed:
chore(ajv): had to disable strict validation no idea what the origin of the issue is

0x4007 committed:
fix(ajv): loading default config is stable

0x4007 committed:
chore: this is all broken

EtherealGlow commented:
I pray for the refactor to be done quickly 🙏

whilefoo committed:
Merge remote-tracking branch 'pavlovcik/refactor/general' into custom-context

0x4007 committed:
Merge pull request #29 from ubiquity-whilefoo/custom-context

feat: custom context

0x4007 committed:
feat: contributor incentives

0x4007 committed:
feat: contributor incentives

0x4007 committed:
Merge remote-tracking branch 'pavlovcik/refactor/general' into feat/contributor-incentives

0x4007 committed:
revert: undo context merges

0x4007 committed:
chore: add context and hold off on full incentives scoring

0x4007 committed:
chore: add context and hold off on full incentives scoring

0x4007 committed:
refactor: try and calculate main scores

0x4007 committed:
refactor(scoring): trying to consolidate into single normalization and calculation

0x4007 commented:
Notes to self on incentive calculation architecture:

- work backwards to determine essential data (design incentive tables for every type first)
- consider firing off events to decouple the code

0x4007 committed:
chore: create permit comment mockup and interfaces

0x4007 committed:
fix(ms): safe ms wrapper because does not support months

0x4007 committed:
feat: aggregate errors

0x4007 committed:
Delete .github/ISSUE_TEMPLATE/sweep-template.yml

0x4007 committed:
fix: wildcard unassigns

and temporarily suppress some build errors

0x4007 committed:
feat: allow setting price label only

also will automatically choose the lowest price label if multiple are set

whilefoo committed:
feat: tried zod

whilefoo committed:
Merge remote-tracking branch 'pavlovcik/refactor/general' into refactor/bot-config

EtherealGlow commented:
Do you guys have any estimate on when the refactor will be done

whilefoo committed:
feat: new config

whilefoo committed:
fix: circular dependencies

whilefoo committed:
fix: schema and decode

whilefoo committed:
fix: payload user type

0x4007 committed:
feat: contributor incentives total scoring (unstable)

0x4007 commented:
I think next week. It feels really close to being done but I'm adding in some time for unknowns during the rollout which I feel is inevitable.

0x4007 committed:
fix: main type errors

0x4007 committed:
refactor: generalizing the contribution scoring schema

0x4007 committed:
feat: comment scoring cleaned up and includes more information

0x4007 committed:
fix: almost at the end of making better types for scoring

whilefoo committed:
fix: transform/decode, config validation at the start

0x4007 committed:
fix: got to permit generation

doing another pass to add a missing value

0x4007 committed:
fix: type errors

sorting based on issue comments or review comments

0x4007 committed:
fix: permit type errors

0x4007 committed:
fix: type errors almost completed

just need to solve setting relevance score correctly

0x4007 committed:
fix: all type errors fixed

0x4007 committed:
feat(start): works with only price set

whilefoo committed:
feat: error messages, probot context in adapters

wannacfuture committed:
feat: initialize

0x4007 committed:
fix: some runtime bugs

0x4007 committed:
Merge remote-tracking branch 'pavlovcik/feat/contributor-incentives-total-scoring-initialization' into feat/contributor-incentives-total-scoring

0x4007 committed:
refactor: cleanup

whilefoo committed:
fix: tests

EtherealGlow commented:
Any new estimate for how much time is left?

whilefoo committed:
fix: added comment

0x4007 committed:
refactor: small adjustments

0x4007 committed:
Merge branch 'refactor/bot-config' of https://github.com/ubiquity-whilefoo/ubiquibot into ubiquity-whilefoo-refactor/bot-config

0x4007 committed:
refactor: organize imports

0x4007 committed:
Merge branch 'ubiquity-whilefoo-refactor/bot-config' into feat/contributor-incentives-total-scoring

0x4007 committed:
fix: build errors

0x4007 committed:
fix(config): token limit moved to misc

0x4007 committed:
Delete .github/ubiquibot-config.yml

0x4007 committed:
fix(config): loader still not stable

0x4007 committed:
refactor: cleanup

whilefoo committed:
feat: remove zod

whilefoo committed:
fix: merge before validation/defaults
reference: https://github.com/pavlovcik/ubiquibot/pull/38#issuecomment-1810016243

whilefoo committed:
feat: custom merge for arrays and change commands to disabledCommands
also changed time and priority labels from array of objects to array of strings

0x4007 committed:
Merge pull request #45 from ubiquity-whilefoo/fix-config

Fix config

0x4007 committed:
feat: logs do not render in console or comment if filtered out based on log level

0x4007 committed:
feat: logs do not render in comment if filtered out and if bubbled to top via return etc

0x4007 committed:
fix: refactored 'ubiquibot-config.yml' and redesigned logger statements

0x4007 committed:
feat: comment on commit

0x4007 committed:
feat: improved comment posting and structured metadata util

wannacfuture committed:
feat: duplication fix

wannacfuture committed:
feat: minor fix

wannacfuture committed:
feat: minor fix

0x4007 committed:
Merge pull request #48 from pavlovcik/feat/contributor-incentives-total-scoring-duplicate-fix

0x4007 committed:
refactor: type property to className

0x4007 committed:
Merge branch 'feat/contributor-incentives-total-scoring' of https://github.com/pavlovcik/ubiquibot into feat/contributor-incentives-total-scoring

wannacfuture committed:
feat: duplication collaborator

0x4007 committed:
Merge pull request #50 from wannacfuture/feat/contributor-incentives-total-scoring-duplicate-fix

feat: duplication collaborator

whilefoo committed:
feat: remove context from addapters and pass it by parameter

whilefoo committed:
fix: change to undefined

whilefoo committed:
fix: initialize only once

whilefoo committed:
feat: logger inside context

0x4007 committed:
chore: delete ubiquibot-config-default.ts and modify
generateConfiguration function

https://github.com/pavlovcik/ubiquibot/commit/ca1d827aa45c16f41e856040fc65a38d5145f1ab#r132819581

0x4007 committed:
Merge pull request #51 from ubiquity-whilefoo/runtime-fix

feat: remove context from adapters and pass it by parameter

0x4007 committed:
fix: logs constructor arguments order

wannacfuture committed:
feat: integrate infrastructure

wannacfuture committed:
Merge branch 'feat/contributor-incentives-total-scoring' into feat/contributor-incentives-total-scoring-duplicate-fix

wannacfuture committed:
feat: update

wannacfuture committed:
feat: update

EtherealGlow commented:
🙏

wannacfuture committed:
feat: removep package lock

0x4007 committed:
fix: await promise

whilefoo committed:
feat: replace issue helpers with paginate

wannacfuture committed:
feat: minor fix

wannacfuture committed:
Merge branch 'feat/contributor-incentives-total-scoring-duplicate-fix' of https://github.com/pavlovcik/ubiquibot into feat/contributor-incentives-total-scoring-duplicate-fix

0x4007 committed:
fix: linting issues and improve code readability

0x4007 committed:
fix: linting issues and improve code readability

0x4007 committed:
fix: linter issues

0x4007 committed:
fix: lint staged

0x4007 committed:
fix: import paths in files

0x4007 committed:
refactor: remove unnecessary barrelling

0x4007 committed:
feat: kebab-case check in pre-commit

whilefoo committed:
feat: add paginate and catch everywhere

0x4007 committed:
Merge pull request #53 from ubiquity-whilefoo/paginate-fix

Octokit paginate

0x4007 committed:
refactor: remove unused exports and functions

0x4007 committed:
Merge branch 'feat/contributor-incentives-total-scoring-duplicate-fix' into refactor/type-locations

0x4007 committed:
Merge remote-tracking branch 'pavlovcik/feat/contributor-incentives-total-scoring' into refactor/type-locations

0x4007 committed:
Merge branch 'refactor/type-locations' into feat/contributor-incentives-total-scoring

0x4007 committed:
Merge branch 'feat/contributor-incentives-total-scoring' into refactor/general

0x4007 committed:
chore: sync readme

0x4007 committed:
Merge branch 'development' into refactor/general

0x4007 committed:
fix: add jest types again

0x4007 committed:
fix: kebab case

0x4007 committed:
fix: logger

0x4007 committed:
feat: convert errors to fatal and warn to errors because seems less ambiguous

0x4007 committed:
chore: update push event logging messages

0x4007 committed:
fix: log level and error handling

on the github comment it will make the metadata fully visible on fatals not just on errors now

0x4007 committed:
revert: this needs type information

0x4007 committed:
fix: default branch in dispatchWorkflow

0x4007 committed:
chore: kebab case

0x4007 committed:
refactor: issueNumber

gitcoindev committed:
chore: add knip ci workflow

Resolves: #897

gitcoindev committed:
chore: add knip.ts configuration skeleton

Resolves: #897

wannacfuture commented on src/bindings/event.ts:
```diff
@@ -1,146 +1,266 @@
+    if (payload.action === "issueClosed") {
+      //This is response for issueClosed request
+      const response = dispatchPayload.client_payload.result;
+      if (response.comment as Comment) {
+        const uncompressedComment = zlib.gunzipSync(Buffer.from(response.comment));
```
```suggestion
      if (response) {
        const uncompressedComment = zlib.gunzipSync(Buffer.from(response));
```

wannacfuture replied:
Seems config repo returns response as the comment string.

0x4007 committed:
Update event.ts

Co-authored-by: FUTURE <danddycandy@gmail.com>

wannacfuture committed:
feat: update

wannacfuture committed:
Update webhooks.ts

wannacfuture committed:
feat: update

wannacfuture committed:
Merge branch 'refactor/general' of https://github.com/wannacfuture/ubiquibot into refactor/general

0x4007 commented:
@wannacfuture its better that you branch from here instead of pushing to mine in case I want to work.

EtherealGlow commented:
When will the refactor be done 😭

wannacfuture committed:
feat: update

wannacfuture committed:
feat: update

wannacfuture committed:
feat: update

wannacfuture committed:
feat: update

wannacfuture committed:
feat: update

wannacfuture committed:
feat: add app deploy documentation

wannacfuture committed:
feat: delete

wannacfuture committed:
feat: update

0x4007 committed:
Apply suggestions from code review

0x4007 committed:
Merge pull request #58 from wannacfuture/refactor/general-deploy

Refactor/general deploy

0x4007 committed:
fix: dispatch params

0x4007 committed:
Revert "Refactor/general deploy"

0x4007 committed:
fix: netlify commit hash

0x4007 committed:
Merge pull request #59 from pavlovcik/revert-58-refactor/general-deploy

Revert "Refactor/general deploy"

0x4007 committed:
fix: load env

0x4007 committed:
fix: netlify commit hash

0x4007 committed:
fix(build): fix some build errors

0x4007 committed:
chore: hack to get this into production faster

0x4007 committed:
fix: commit hash render on build

0x4007 committed:
fix: commit command

gitcoindev committed:
Update knip.yml to use knip-reporter v2.1.2 and node 20.x

0x4007 committed:
Merge branch 'feat/logger-fatal' into refactor/general

0x4007 committed:
fix: delegate compute message level

0x4007 committed:
fix: ok log level to info level

0x4007 committed:
fix: minor missing info

0x4007 committed:
chore: webhook cleaning

0x4007 committed:
fix: build hash

0x4007 committed:
fix: lint-staged

0x4007 committed:
fix: compute repo name

0x4007 committed:
fix(logs): always log to supabase

0x4007 committed:
refactor: remove unused log levels

0x4007 commented on netlify.toml:
```diff
@@ -1,6 +1,5 @@
-# Specify is-core-module/package.json as an external dependency
-ignore = "is-core-module/package.json"
\ No newline at end of file
+  command = "echo \"export const COMMIT_HASH = '$COMMIT_REF';\" > src/commit-hash.ts"
```
This is a hack to access the current commit hash only during Netlify build time, and saving it to be loaded in the runtime later.

0x4007 commented on src/handlers/comment/handlers/assign/generate-assignment-comment.ts:
```diff
@@ -0,0 +1,43 @@
+    deadline,
+    registeredWallet:
+      (await runtime.adapters.supabase.wallet.getAddress(payload.sender.id)) ||
+      "Please set your wallet address to use `/wallet 0x0000...0000`",
```
We should throw an error to force them to set a wallet address before self assignment.

0x4007 committed:
chore: debug validation error

0x4007 committed:
fix: unassign handler but root cause not diagnosed

gitcoindev committed:
feat: switch logging to ubiquibot-logging NPM module

gitcoindev committed:
feat: enable pretty logs import with ubiquity-logger v0.3.0

0x4007 committed:
Merge pull request #60 from gitcoindev/refactor-general-with-npm-logging

0x4007 committed:
refactor: move to delegated compute

whilefoo commented:
I'm trying to use the new version with workflow dispatcher and I'm getting an error: `{"message":"inputs are too large.","documentation_url":"https://docs.github.com/rest/actions/workflows#create-a-workflow-dispatch-event"}`. 

It's trying to pass all comments from the issue and PR so for large issues this is gonna be a problem.
@wannacfuture

0x4007 committed:
refactor: prefix types with github to avoid typename collisions

0x4007 commented:
> I'm trying to use the new version with workflow dispatcher and I'm getting an error: `{"message":"inputs are too large.","documentation_url":"https://docs.github.com/rest/actions/workflows#create-a-workflow-dispatch-event"}`.
> 
> It's trying to pass all comments from the issue and PR so for large issues this is gonna be a problem. @wannacfuture

I asked them to fix this. We will take a new approach. We will pass in minimal information (repository name, organization name, issue number) and then fetch the data from the GitHub API instead on the delegated compute side. 

I got started on it over at 

https://github.com/pavlovcik/ubiquibot-config/compare/feat/pass-less-info-in?expand=1
https://github.com/pavlovcik/ubiquibot/compare/refactor/general...pavlovcik:ubiquibot:refactor/move-to-delegated-compute

gitcoindev committed:
chore: switch knip-reporter to latest upstream release

Version 2.1.0 released with export annotations support and additional
tests.

gitcoindev committed:
chore: set node LTS and knip v2 tag

whilefoo committed:
feat: nft rewards

whilefoo committed:
fix: remove console log

whilefoo commented:
There also seems to be an issue with ubiquibot-logger

```
const ubiquibot_logger_1 = require("ubiquibot-logger");
                           ^
Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/whilefoo/ubiquity/ubiquibot/node_modules/ubiquibot-logger/dist/logs.js from /Users/whilefoo/ubiquity/ubiquibot/dist/adapters/adapters.js not supported.
Instead change the require of logs.js in /Users/whilefoo/ubiquity/ubiquibot/dist/adapters/adapters.js to a dynamic import() which is available in all CommonJS modules.
```

gitcoindev commented:
> There also seems to be an issue with ubiquibot-logger
> 
> ```
> const ubiquibot_logger_1 = require("ubiquibot-logger");
>                            ^
> Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/whilefoo/ubiquity/ubiquibot/node_modules/ubiquibot-logger/dist/logs.js from /Users/whilefoo/ubiquity/ubiquibot/dist/adapters/adapters.js not supported.
> Instead change the require of logs.js in /Users/whilefoo/ubiquity/ubiquibot/dist/adapters/adapters.js to a dynamic import() which is available in all CommonJS modules.
> ```

Hi @whilefoo , the `ubiquibot-logger` is an ESM module https://nodejs.org/api/esm.html#modules-ecmascript-modules which is now a default, please import the module using `import` statement instead of `require` and the error should be gone.

https://medium.com/@chamin.njay/require-vs-import-in-node-js-abdf5427d7b0

> When to Use Which?
> If you’re building a new project or working on the front-end, prefer using import for its modern features and static analysis capabilities.
> If you’re maintaining an older Node.js application or need to work in an environment where ES Modules aren’t well-supported, stick with require.

gitcoindev committed:
chore: use Codex-/knip-reporter

whilefoo commented:
> Hi @whilefoo , the `ubiquibot-logger` is an ESM module https://nodejs.org/api/esm.html#modules-ecmascript-modules which is now a default, please import the module using `import` statement instead of `require` and the error should be gone.

it is imported with `import`
https://github.com/ubiquity/ubiquibot/blob/3ca8fada16fcfef43b3a8409d1b4f7e343afda4c/src/adapters/adapters.ts#L5

The problem is probably because `ubiquibot-logger` package is not converted to CommonJS because TS transpiles our code from ESM to CommonJS

gitcoindev commented:
> > Hi @whilefoo , the `ubiquibot-logger` is an ESM module https://nodejs.org/api/esm.html#modules-ecmascript-modules which is now a default, please import the module using `import` statement instead of `require` and the error should be gone.
> 
> it is imported with `import`
> 
> https://github.com/ubiquity/ubiquibot/blob/3ca8fada16fcfef43b3a8409d1b4f7e343afda4c/src/adapters/adapters.ts#L5
> 
> The problem is probably because `ubiquibot-logger` package is not converted to CommonJS because TS transpiles our code from ESM to CommonJS

I see, I will check today / tomorrow if this module can be shipped as a both CommonJS and ESM package and update it.

0x4007 committed:
refactor: cleanup

0x4007 committed:
fix: app permissions used instead of personal access token

gitcoindev commented:
> > > Hi @whilefoo , the `ubiquibot-logger` is an ESM module https://nodejs.org/api/esm.html#modules-ecmascript-modules which is now a default, please import the module using `import` statement instead of `require` and the error should be gone.
> > 
> > 
> > it is imported with `import`
> > https://github.com/ubiquity/ubiquibot/blob/3ca8fada16fcfef43b3a8409d1b4f7e343afda4c/src/adapters/adapters.ts#L5
> > 
> > The problem is probably because `ubiquibot-logger` package is not converted to CommonJS because TS transpiles our code from ESM to CommonJS
> 
> I see, I will check today / tomorrow if this module can be shipped as a both CommonJS and ESM package and update it.

Hi @whilefoo I checked in `tsconfig.json` of the ubiquibot-logger module, `esModuleInterop` is enabled:

"esModuleInterop": true /* Enables emit interoperability between CommonJS and ES Modules via creation of namespace objects for all imports. Implies 'allowSyntheticDefaultImports'. */,

Following the article https://dev.to/tigawanna/building-and-publishing-npm-packages-with-typescript-multiple-entry-points-tailwind-tsup-and-npm-9e7 I enabled 'require' and 'import' in tsup exports and published v0.3.1 , let's check if this solves the import error.

0x4007 committed:
refactor: remove unnecessary code

0x4007 committed:
chore: add debugging logs

0x4007 committed:
refactor: rename assign to start and unassign to stop

0x4007 committed:
chore: debugging

0x4007 committed:
chore: debugging

0x4007 committed:
chore: debugging

0x4007 committed:
chore: debugging

0x4007 committed:
chore: debugging add netlify cli

0x4007 committed:
chore: debugging

0x4007 committed:
feat(deps): add netlify logs

0x4007 committed:
chore(logs): add metadata always

0x4007 committed:
chore: enable source maps for netlify webhook for debugging

0x4007 committed:
chore: remove huge log

0x4007 committed:
fix: type error for build

0x4007 committed:
chore: check necessary environment variables

gitcoindev committed:
chore: upgrade knip to v3.7 with dev depencies typescript 5.0.4 and
node types

Also set explicit anntations in knip.yml.

0x4007 committed:
refactor: break apart unassign logic

0x4007 committed:
fix: continue searching for assignees to disqualify

0x4007 committed:
refactor: rename files

0x4007 committed:
refactor: move functions to where they are used

0x4007 committed:
chore: add penalty placeholder for issue reopened

0x4007 committed:
feat: check if wallet is set before allowing start

0x4007 committed:
feat: add more github events to enum

0x4007 committed:
fix: build error

0x4007 committed:
feat: extracted every github webhook type from probot

0x4007 committed:
fix: build errors

whilefoo commented:
> Following the article https://dev.to/tigawanna/building-and-publishing-npm-packages-with-typescript-multiple-entry-points-tailwind-tsup-and-npm-9e7 I enabled 'require' and 'import' in tsup exports and published v0.3.1 , let's check if this solves the import error.

I tried v0.3.1 but it's still the same error

gitcoindev commented:
hi @whilefoo I will give this priority and try to fix till Monday EOD.

0x4007 commented:
You guys should make sure you're using the same version of Node. This is the only environment related concern that I can think of to explain why it works for gitcoindev and not whilefoo. It looks like we first must identify which version works and then enforce it here:

https://github.com/ubiquity/ubiquibot/blob/c015fc791afdec32856d1c30855273b05b587a86/package.json#L88

For CI purposes I set the node version to LTS. @gitcoindev you should strive to make this work with this version of node.

https://github.com/ubiquity/ubiquibot/blob/development/.nvmrc#L1

wannacfuture committed:
feat: pass less info

gitcoindev commented:
@whilefoo I am on v0.3.4 already, so far no luck but closer and understand now where it fails as I can reproduce easily now on E2E tests on my fork's experimental branch based on refactor/general. I will continue tomorrow on this and try to provide a proper solution.

0x4007 committed:
refactor: move github events

0x4007 committed:
Merge branch 'refactor/general-deploy' into refactor/move-to-delegated-compute

0x4007 committed:
chore: hack for probot to authenticate

0x4007 committed:
chore: rename APP_PRIVATE_KEY to PRIVATE_KEY for probot

0x4007 committed:
fix: dispatcher use current organization

0x4007 committed:
chore: debugging duplicate events

0x4007 committed:
fix: debugging duplicate events

0x4007 committed:
fix: double events

0x4007 committed:
fix: build

0x4007 committed:
fix: build

gitcoindev commented:
hi @whilefoo could you please check ubiquibot-logger v0.3.5  ? I switched to cjs module format in tsup config, the same format as in ubiquibot's tsconfig.json `"module": "commonjs"` setting and I no longer see the import error in my fork's E2E tests. If you can still reproduce the error with v0.3.5, please guide me how to reproduce and/ or find the error logs and I will be happy to help and fix this.

gitcoindev commented:
Btw, about @pavlovcik 's last comment, I agree that we should sync all repositories to use v20.3.0 LTS node version if this was selected for development and use the same version both in workspace settings and GitHub actions if possible. Just a side note: for ubiquibot-logger I checked v20.3.0 and v20.10.0.

0x4007 committed:
chore: upgrade yarn

0x4007 committed:
fix: ping assignees for deadline

0x4007 committed:
refactor: remove redundant value organization

0x4007 committed:
feat: dedicated plugin stored at ubiquibot org

0x4007 committed:
feat: enable repository authentication from remote via github app

0x4007 committed:
refactor: type casting for installation id

whilefoo committed:
Merge remote-tracking branch 'pavlovcik/refactor/general' into nft-rewards

whilefoo committed:
feat: update ubiquibot logger and fix start command

whilefoo commented:
@gitcoindev it's working now, thanks!

0x4007 committed:
Merge pull request #57 from gitcoindev/knip-ci

wannacfuture committed:
feat: fix pricing issue

0x4007 committed:
Merge pull request #63 from pavlovcik/refactor/general-deploy-pricing-issue

feat: fix pricing issue

wannacfuture committed:
feat: remove private key from netlify env

0x4007 committed:
Merge pull request #64 from wannacfuture/refactor/remove-25519key-from-netlify

0x4007 committed:
chore: almost done merging

0x4007 committed:
chore: merge

0x4007 committed:
Merge branch 'ubiquity-whilefoo-nft-rewards' into refactor/move-to-delegated-compute

wannacfuture committed:
feat: fix duplication price label issue

0x4007 committed:
Merge pull request #65 from wannacfuture/feat/fix-label-issues

wannacfuture committed:
feat: fix build errors

wannacfuture committed:
feat: fix CI erros

wannacfuture committed:
Update build.yml

wannacfuture committed:
Update build.yml

wannacfuture committed:
feat: minor change

wannacfuture committed:
Update build.yml

wannacfuture committed:
feat: update build.yml

wannacfuture committed:
Update build.yml

wannacfuture committed:
feat: update build.yml

wannacfuture committed:
Update build.yml

wannacfuture committed:
feat: update

0x4007 committed:
Merge pull request #67 from wannacfuture/feat/fix-label-issues

Feat/fix label issues

wannacfuture committed:
feat: remove axios call

0x4007 committed:
Merge pull request #68 from wannacfuture/refactor/remove-axios-call

feat: remove axios call

0x4007 committed:
Revert "feat: remove axios call"

0x4007 committed:
Merge pull request #69 from pavlovcik/revert-68-refactor/remove-axios-call

Revert "feat: remove axios call"

0x4007 committed:
Merge branch 'refactor/move-to-delegated-compute' into development

0x4007 committed:
Merge remote-tracking branch 'upstream/development' into HEAD

0x4007 committed:
Merge branch 'development' into refactor/general

0x4007 committed:
chore: migrate to bun

0x4007 committed:
fix: bun version

0x4007 committed:
fix: build command

0x4007 committed:
fix: build command

0x4007 committed:
fix: install toolchain

0x4007 commented:
@gitcoindev maybe you can help fix the knip CI here

0x4007 committed:
chore: lint errors

0x4007 committed:
chore: try to fix knip

0x4007 committed:
chore: adjustments

BeanieMen commented:
🎉
